### ISSUE TITLE
Investigate spring plugin upgrade failure

### ISSUE THREAD

**Issue by thetaurean** [CDCgov/prime-reportstream#15909]
This is low priority until the microservice / submission receiver epic starts moving again.

### User Story
As a developer I want to utilize the latest version of dependencies to take advantage of the latest bugfixes, security patches, and possibly features.

### Description/Use Case
During a recent dependabots push the spring plugin (#15696) caused the destination filter function to fail during execution. More details in dev notes.

### Risks/Impacts/Considerations
- Compatibility with the spring framework dependency

### Dev Notes
Exception prevented the destination filter from running at all. It was pertaining to `QueueMessage` being cast into the `FhirDestinationQueueMessage` type. It can't find the subtype for `destination-filter` so I suspect something changed related to the `registerPrimeRouterQueueMessageSubtypes` function.

### Acceptance Criteria
- Destination filter and remainder of pipeline runs successfully with new version of spring plugin
- Underlying issue and its fix is documented somehow (could be in PR)

**Comment by JFisk42** [comment]
Will revisit this ticket once blocking story is complete. @david-navapbc

**Comment by thetaurean** [comment]
A newer version of the dependency resolved the test failure so this investigation is no longer needed.
